### ISSUE TITLE
Unnecessary <pre> tags created in HTML

### ISSUE THREAD

**Issue by supetronix** [DioxusLabs/dioxus#500]
**Problem:**
1) While using "conditional rendering" in web, it generates "empty pre" tags which seems odd. I checked "yew" and it doesn't happen there.

2) Perhaps it is just me but "rsx! { }" and "rsx!( )" requirements are not consistent (for nested use within conditional rendering).

3) In "yew" you can only provide an "if" condition but in "Dioxus" you must provide "else" clause too, unless you use ".then", there must be a good reason for this but it just seems unnecessary code.

_I am new to both Yew and Dioxus, although not to development, so I apologize if I missed something and unknowingly opened an unnecessary defect._

**Steps To Reproduce**

1. Attaching two screenshots. First one is for "Dioxus" where in HTML you will see "pre" tags, when "if" condition doesn't match and else part executes, which returns "none". In 2nd screenshot, from "yew", similar if condition is tested and it doesn't create extra "pre" tags.

2. In "Dioxus" (first screenshot) you will see that line "11" can only work if I use "rsx! (cx, ...)" while code in "if let some" block (line 4 just below there) can work with just "rsx!{ } ".

**Expected behavior**

1. No "pre" tags are created when we are conditionally rendering HTML and don't want to output something.
2. Either "rsx!{}" is allowed nested in conditional "if" or not allowed at all. It will be difficult to remember when which variant can be used, IMHO.

**Screenshots**

1st (Dioxus):
<img width="1058" alt="Dioxus" src="https://user-images.githubusercontent.com/17693921/179154757-23836793-3e19-4bd7-a19c-f93fbd8bffe8.png">

2nd (Yew):

<img width="1253" alt="Yew" src="https://user-images.githubusercontent.com/17693921/179154785-0d20c592-d055-40c2-ab6d-1ab5d49831ea.png">

**Environment:**
 - Dioxus version: 0.2.4
 - Rust version: 1.62
 - OS info: MacOS
 - App platform: Web

**Comment by RubixDev** [comment]
As mentioned in https://github.com/nissy-dev/dioxus-free-icons/issues/14, this is the current explanation:

https://github.com/DioxusLabs/dioxus/blob/94004cfe19c1efb851b3d8147f608990c9ab21b5/packages/core/src/diff.rs#L29-L39

**Comment by bishoyroufael** [comment]
I was facing the same issue and did a dirty workaround with JS to remove the `innerHTML` of the element. In case anyone is interested, you can do something like this:
```javascript
// public/index.js

const ELEM_SELECTOR = "#notita-editable-area";

function waitForElm(selector) {
    return new Promise(resolve => {
        if (document.querySelector(selector)) {
            return resolve(document.querySelector(selector));
        }

        const observer = new MutationObserver(mutations => {
            if (document.querySelector(selector)) {
                observer.disconnect();
                resolve(document.querySelector(selector));
            }
        });

        observer.observe(document.documentElement, {
            childList: true,
            subtree: true
        });
    });
}

// Workaround to remove Dioxus inner <pre> tag
waitForElm(ELEM_SELECTOR).then((elem)=>{
    elem.innerHTML = "";
})

```

You can then add this JS to your custom head method `with_custom_head(..)` ([docs](https://docs.rs/dioxus-desktop/latest/dioxus_desktop/struct.Config.html#method.with_custom_head)) 

Example:
```rust
 dioxus_desktop::launch_cfg(App, dioxus_desktop::Config::new().with_custom_head(
             r#"
            <head> 
                <script src="public/index.js"> </script>
            </head>
            "#.to_string()))
```

**Comment by jkelleyrtp** [comment]
Revisiting this from way in the future - I think we can just use comment nodes instead of <pre> elements. Need to be careful that comment nodes don't get merged, but it should work.

**Comment by jkelleyrtp** [comment]
This has been fixed! In #2579
